lib/mongo/bulk_write.rb

```diff
@@ -154,11 +155,39 @@ def single_statement?(operation)
       SINGLE_STATEMENT_OPS.include?(operation.keys.first)
     end
 
+    # Get the write concern for the bulk write.
+    #
+    # If in a transaction and the bulk write inherits an 
+    # unacknowledged write concern from the collection, remove
+    # the write concern's :w option. Otherwise, return the
+    # unmodified write concern.
+    #
+    # @return [ Mongo::WriteConcern ] The write concern.
+    def applied_write_concern(session)
+      if options[:write_concern]
+        WriteConcern.get(options[:write_concern])
+      else 
+        applied_collection_write_concern(session)
+      end
+    end
+
+    def applied_collection_write_concern(session)
+      wc = collection.write_concern
+      if session.respond_to?(:in_transaction?) && session.in_transaction?
+        if wc && WriteConcern.send(:unacknowledged?, wc.options)
+          opts = wc.options.dup
+          opts.delete(:w)
```

Comment:
I believe this should only be done if `w == 0`.

Reply:
The `unacknowledged?` method just checks for `w == 0`, but if you would prefer for that to be explicit here, I can do that.

Reply:
You are right, no changes are needed.